src/kitsuneTools.ts

```diff
@@ -17,12 +26,37 @@ export class KitsuneTools {
         let sig0 = utils.splitSignature(appointment.stateUpdate.signatures[0]);
         let sig1 = utils.splitSignature(appointment.stateUpdate.signatures[1]);
 
-        const tx = await contract.setstate(
-            [sig0.v - 27, sig0.r, sig0.s, sig1.v - 27, sig1.r, sig1.s],
-            appointment.stateUpdate.round,
-            appointment.stateUpdate.hashState
-        );
-        return await tx.wait();
+        try {
+            logger.debug("a1");
+
+            let trying = true;
+            let tries = 0;
+            let tx;
+            while (trying && tries < 10) {
+                try {
+                    tx = await contract.setstate(
+                        [sig0.v - 27, sig0.r, sig0.s, sig1.v - 27, sig1.r, sig1.s],
+                        appointment.stateUpdate.round,
+                        appointment.stateUpdate.hashState
+                    );
+                    trying = false;
+                } catch (exe) {
+                    // lets retry this hard until we can no longer
+                    logger.error(`Failed to set state for contract ${contract.address}, re-tries ${tries}`)
+                    tries++;
+                    await wait(1000);
+                }
+            }
+
+            if(trying) throw new Error("Failed after 10 tries.")
```

Comment:
why not do if (tries > 10) instead of using a separate environment variable

Reply:
good catch - this whole section of code shouldn't have been committed :s

We should be doing retry logic in the watcher, not in the kitsune tools.

Interestingly I had to do this to even get the basic load test to pass. Many events require multiple retries when they're happening concurrently - I think the culprit is ganache.